### PR TITLE
feat: collect metrics for requests

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR introduces the capability to collect metrics for requests in a session

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-61877

### CODE REVIEW

File: ios/NetworkClient.swift [hunk 1]
```diff
@@ -293,4 +364,62 @@ extension NetworkClient {
...
+        monitor.pathUpdateHandler = { path in
+            if path.usesInterfaceType(.wifi) {
+                networkType = "Wi-Fi"
+            } else if path.usesInterfaceType(.cellular) {
+                let telephonyNetworkInfo = CTTelephonyNetworkInfo()
+                if let currentRadioAccessTechnology = telephonyNetworkInfo.serviceCurrentRadioAccessTechnology?.values.first {
+                    if #available(iOS 14.1, *) {
+                        switch currentRadioAccessTechnology {
+                        case CTRadioAccessTechnologyGPRS:
+                            networkType = "2G (GPRS)"
```
Comment (isacikgoz) [inline ios/NetworkClient.swift:383]: Is it possible to run this application on 2G networks?

**Review by isacikgoz — APPROVED** [review]
LGTM overall. wondering if we could classify network regarding to their speed rather than their type?

File: ios/NetworkClient.swift (hunk 1 above)
Comment (enahum) [reply ios/NetworkClient.swift:383]: I don't think it will happen but I see no reason why it shouldn't work

**Comment by enahum** [comment]
> LGTM overall. wondering if we could classify network regarding to their speed rather than their type?

I'll give it a try and see if I can add it, but no promises

File: src/APIClient/NativeApiClient.ts [hunk 2]
```diff
@@ -6,6 +6,18 @@ import type {
...
+    networkType: string;
+    tlsCipherSuite: string;
+    tlsVersion: string;
+    httpVersion: string;
+    isCached: boolean;
+    compressedSize: Double;
+    size: Double;
+    connectionTime: Double;
+    latency: Double;
+};
```
Comment (rahimrahman) [inline src/APIClient/NativeApiClient.ts:19]: Shouldn't this be the same as APIClient.ts?  Can't we define here and re-export there?

Missing at least `speedInMbps`, maybe?

**Review by rahimrahman — APPROVED** [review]
I must admit, I asked chatgpt to help review this and couldn't find anything worth bringing up.  I used your branch for testing with multiple scenarios (increasing latency).

I only have 1 comment about missing fields in a type in which I think could be the same for both for codegen and APIclient.

File: src/APIClient/NativeApiClient.ts (hunk 2 above)
Comment (enahum) [reply src/APIClient/NativeApiClient.ts:22]: fixed

**Pull request merged**